chatkit-android/src/main/kotlin/com/pusher/chatkit/AndroidChatkitDependencies.kt

```diff
@@ -1,33 +1,17 @@
 package com.pusher.chatkit
 
 import android.content.Context
-import com.pusher.SdkInfo
-import com.pusher.platform.AndroidDependencies
-import com.pusher.platform.PlatformDependencies
+import com.pusher.platform.AndroidMediaTypeResolver
+import com.pusher.platform.MediaTypeResolver
 import com.pusher.platform.logger.Logger
 import com.pusher.platform.tokenProvider.TokenProvider
-import okhttp3.OkHttpClient
 
-/**
- * [ChatkitDependencies] implementation for Android using [AndroidDependencies] to fulfil [PlatformDependencies].
- */
-data class AndroidChatkitDependencies @JvmOverloads constructor(
-    override val tokenProvider: TokenProvider,
-    override val okHttpClient: OkHttpClient? = null,
-    private val platformDependencies: PlatformDependencies = AndroidDependencies(),
-    override val logger: Logger = platformDependencies.logger,
-    private val context: Context? = null
-) : ChatkitDependencies {
+class AndroidChatkitDependencies @JvmOverloads constructor(
+    val context: Context,
+    tokenProvider: TokenProvider,
+    override val logger: Logger = EmptyLogger(),
+    override val mediaTypeResolver: MediaTypeResolver = AndroidMediaTypeResolver()
+) : ChatkitDependencies(tokenProvider) {
 
-    override val mediaTypeResolver = platformDependencies.mediaTypeResolver
-    override val appHooks = AndroidAppHookEmitter()
-    override val pushNotifications =
-            if (context != null) BeamsPushNotificationsFactory(context) else null
-
-    override val sdkInfo = SdkInfo(
-            product = "Chatkit",
-            sdkVersion = BuildConfig.VERSION_NAME,
-            platform = "Android",
-            language = "Kotlin/Java"
-    )
+    override val sdkInfo = super.sdkInfo.copy(platform = "Android")
```

Comment:
should this be a hardcoded string here - would it be better as a const at the top of the file?

Reply:
Usage is clear (it's clear what the value means) and its a single usage. Extracting a const would introduce unnecessary indirection (just by looking at the usage one would wonder what actual value is).